packages/vouching/contracts/Vouching.sol

```diff
@@ -0,0 +1,101 @@
+pragma solidity ^0.4.24;
+
+
+import "./ZepToken.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+contract Vouching {
+  using SafeMath for uint256;
+
+  struct Dependency {
+    address owner;
+    address dependencyAddress;
+    uint256 stake;
+  }
+
+  mapping (string => Dependency) private _registry;
+  uint256 private _minimumStake;
+  ZepToken private _token;
+
+  modifier onlyDependencyOwner(string name) {
+    require(msg.sender == _registry[name].owner);
+    _;
+  }
+
+  event OwnershipTransferred(address indexed oldOwner, address indexed newOwner);
+  event DependencyCreated(string name, address indexed owner, address indexed dependencyAddress, uint256 initialStake);
+  event Vouched(string name, uint256 amount);
+  event Unvouched(string name, uint256 amount);
+  event DependencyRemoved(string name);
+
+  constructor(uint256 minimumStake, ZepToken token) public {
+    require(token != address(0));
```

Comment:
Even if they are not yet supported by truffle, let's add revert reasons to all `require`s.